elm/Main.elm

```diff
@@ -26,15 +29,24 @@ view model =
     in
         div []
             [ ol [] (List.map viewSet sets)
+            , button [ onClick AddSet ] [ text "Add set" ]
             ]
 
 viewSet : (Int, SingleSet.Model) -> Html Msg
 viewSet (i, setModel) =
-    li [] [map (SingleSetMsg << \m -> (i, m)) (SingleSet.view setModel)]
+    li []
+        [ map (SingleSetMsg << \m -> (i, m)) (SingleSet.view setModel)
+        , button [onClick (DeleteSet i)] [ text "Delete" ]
+        ]
 
 update : Msg -> Model -> Model
 update msg model =
     case msg of
+        AddSet ->
+            { model | setModels = model.setModels ++ [SingleSet.initialModel] }
+        DeleteSet i ->
+            { model | setModels = List.removeAt i model.setModels }
```

Comment:
I'd be careful about using position in a list as an ID. If an element's position changes, then so does it's "id". You may want a `Dict` to allow random access by some key.